### PR TITLE
fetch-mock: Add type for matching headers.

### PR DESCRIPTION
**Pull request by vicros0716**
Please fill in this template.

- [x] Use a meaningful title for the pull request. Include the name of the package modified.
- [x] Test the change in your own code. (Compile and run.)
- [x] Follow the advice from the [readme](https://github.com/DefinitelyTyped/DefinitelyTyped/blob/master/README.md#make-a-pull-request).
- [x] Avoid [common mistakes](https://github.com/DefinitelyTyped/DefinitelyTyped/blob/master/README.md#common-mistakes).
- [x] Run `npm run lint package-name` (or `tsc` if no `tslint.json` is present).

If changing an existing definition:
- [x] Provide a URL to documentation or source code which provides context for the suggested changes: [Documentation](http://www.wheresrhys.co.uk/fetch-mock/api) & [source code](https://github.com/wheresrhys/fetch-mock/blob/b8270640d5711feffb01d1bf85bb7da95179c4de/src/fetch-mock.js#L198)
- [x] Increase the version number in the header if appropriate. (Not appropriate)
- [x] If you are making substantial changes, consider adding a `tslint.json` containing `{ "extends": "dtslint/dt.json" }`. (Already exists)

### CODE REVIEW

**Review by tamird — APPROVED** [review]
:+1: would be nice if links to source/documentation were in the commit message.

**Comment by RyanCavanaugh** [comment]
Approved by a listed owner. PR ready to merge pending express review by a maintainer.

**Comment by vicros0723** [comment]
@tamird You're totally right. Sorry I missed on that.

**Pull request merged**
